testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
+    {
+        onTrino().executeQuery("DROP TABLE IF EXISTS test_from_utc_timestamp_source");
+        onHive().executeQuery("CREATE TABLE test_from_utc_timestamp_source AS SELECT " +
+                "  CAST(123 AS tinyint) source_tinyint, " +
+                "  CAST(10123 AS smallint) source_smallint, " +
+                "  CAST(2592000123 AS integer) source_integer, " +
+                "  CAST(2592000123 AS bigint) source_bigint, " +
+                "  CAST(2592000.0 AS float) source_float, " +
+                "  CAST(2592000.123 AS double) source_double, " +
+                "  CAST(2592000.123 AS decimal(10,3)) source_decimal_three," +
+                "  CAST(2592000 AS DECIMAL(10,0)) source_decimal_zero," +
+                "  timestamp '1970-01-30 16:00:00' source_timestamp, " +
+                "  date '1970-01-30 16:00:00' source_date ");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS test_from_utc_timestamp_view");
+        onHive().executeQuery("CREATE VIEW " +
+                "test_from_utc_timestamp_view " +
+                "AS SELECT " +
+                // TODO: TZ identifiers like 'PST' is not supported by Trino; possibly we need a translation map (???)
+                "   from_utc_timestamp(source_tinyint, 'America/Los_Angeles') ts_tinyint, " +
+                "   from_utc_timestamp(source_smallint, 'America/Los_Angeles') ts_smallint, " +
+                "   from_utc_timestamp(source_integer, 'America/Los_Angeles') ts_integer, " +
+                "   from_utc_timestamp(source_bigint, 'America/Los_Angeles') ts_bigint, " +
+                "   from_utc_timestamp(source_float, 'America/Los_Angeles') ts_float, " +
+                "   from_utc_timestamp(source_double, 'America/Los_Angeles') ts_double, " +
+                "   from_utc_timestamp(source_decimal_three, 'America/Los_Angeles') ts_decimal_three, " +
+                "   from_utc_timestamp(source_decimal_zero, 'America/Los_Angeles') ts_decimal_zero, " +
+                "   from_utc_timestamp(source_timestamp, 'America/Los_Angeles') ts_timestamp, " +
+                "   from_utc_timestamp(source_date, 'America/Los_Angeles') ts_date " +
+                "FROM test_from_utc_timestamp_source");
+
+        // check result on Trino
+        assertThat(query("SELECT " +
+                "  CAST(ts_tinyint as VARCHAR), " +
+                "  CAST(ts_smallint as VARCHAR), " +
+                "  CAST(ts_integer as VARCHAR), " +
+                "  CAST(ts_bigint as VARCHAR), " +
+                "  CAST(ts_float as VARCHAR), " +
+                "  CAST(ts_double as VARCHAR), " +
+                "  CAST(ts_decimal_three as VARCHAR), " +
+                "  CAST(ts_decimal_zero as VARCHAR), " +
+                "  CAST(ts_timestamp as VARCHAR), " +
+                "  CAST(ts_date as VARCHAR) " +
+                "FROM test_from_utc_timestamp_view"))
+                .containsOnly(row(
+                        "1969-12-31 16:00:00.123",
```

Comment:
Would be nice to also verify behaviour for very old dates (pre-calendard switch and some negative epoch values).

Reply:
I can try. But I am sure this is not an exact copy of Hive's behavior in some corner cases. Just a best-effort translation.

Reply:
Tests to document the behaviour either way would help. But I don't feel too strongly about this.

Reply:
All the Hive View handling stuff is currently "best effort" and still very "edge case free" at this point.
Let's take care of testing edge cases as a follow up. 
We should also test with double NaN too, besides "historical dates".